### PR TITLE
Use a `RefSet`, rather than a plain Object, for tracking already processed nodes in `PartialEvaluator.hasBlendModes`

### PR DESCRIPTION
**Pull request by Snuffleupagus**

### CODE REVIEW

File: src/core/evaluator.js [hunk 1]
```diff
@@ -322,20 +323,20 @@ class PartialEvaluator {
...
-          processed[xObject.dict.objId] = true;
+          processed.put(xObject.dict.objId);
         }
         var xResources = xObject.dict.get("Resources");
         if (!(xResources instanceof Dict)) {
           continue;
         }
         // Checking objId to detect an infinite loop.
-        if (xResources.objId && processed[xResources.objId]) {
+        if (xResources.objId && processed.has(xResources)) {
```
Comment (timvandermeij) [inline src/core/evaluator.js:333]: I think this should be `processed.has(xResources.objId)`.

Comment (Snuffleupagus) [reply src/core/evaluator.js:333]: Of course it should, I really don't know what happened there; thank you!

**Review by timvandermeij — APPROVED** [review]

**Comment by timvandermeij** [comment]
Thank you for improving this!

**Pull request merged**